src/validation.cpp

```diff
@@ -4356,7 +4356,18 @@ void CChainState::LoadExternalBlockFile(FILE* fileIn, FlatFilePos* dbp)
                     }
                 }
             } catch (const std::exception& e) {
```

Comment:
Thanks for looking at this. I agree, it's likely an overly broad try-catch block and that it's difficult to reason about what types of exceptions might be thrown, and from where, since there is no compile time concept of "checked" exceptions, like in Java. This is why I don't want to try changing it for this PR :-) I'm sure I'll miss something and it'll break at runtime. And, it'll just make the patch larger without impacting the usefulness of this specific fix.

I can see that `blkdat >> block;` will throw at least `std::ios_base::failure` from at least a couple different places within the deserialization code. These are the exceptions that the original issue reported.